src/utils/RealtimeFeedUtils.js

```diff
@@ -3,140 +3,55 @@ import { PYTHON_APP } from './EnvUtils';
- * Return bus and tracking data including location
+ * Given an array of { routeID, tripID },
+ * Return bus information
  * Input:
  [
  {
-   “stopID” : String,
    “routeID” : String,
-   “tripIdentifiers” : [String]
+   tripID : String
  },
  …
  ]
  */
 async function getTrackingResponse(requestData: Object): Object {
   LogUtils.log({ message: 'getTrackingResponse: entering function' });
-
   const trackingInformation = [];
-  const rtf = await fetchRTF(); // ensures the realtimeFeed doesn't update in the middle of execution
+  const vehicles = await fetchVehicles();
 
   // for each input
-  await Promise.all(requestData.map(async (data): Promise<boolean> => {
-    const { stopID, routeID, tripIdentifiers } = data;
-    const realtimeDelayData = getDelayInformation(stopID, tripIdentifiers[0], rtf);
-
-    if (realtimeDelayData) {
-      const authHeader = await TokenUtils.fetchAuthHeader();
-      const options = {
-        method: 'GET',
-        url: 'https://gateway.api.cloud.wso2.com:443/t/mystop/tcat/v1/rest/Vehicles/GetAllVehiclesForRoute',
-        headers: {
-          Authorization: authHeader,
-          'Cache-Control': 'no-cache',
-          'Postman-Token': 'b688b636-87ea-4e04-9f3e-ba34e811e639',
-        },
-        qs: { routeID },
-      };
-
-      const trackingRequest = await RequestUtils.createRequest(options, 'Tracking request failed');
-      if (!trackingRequest) return false;
-
-      /**
-       * Parse request to object and map valid realtime data to info for each bus
-       */
-      const busFound = JSON.parse(trackingRequest) // parse the request
-        .find( // find the first tracking data matching the vehicleId
-          busInfo => busInfo.VehicleId === realtimeDelayData.vehicleId,
-        );
-
-      if (!busFound) return false;
-
-      const trackingData = ((busInfo) => { // create object and return
-        let lastUpdated = busInfo.LastUpdated;
-        const firstParan = lastUpdated.indexOf('(') + 1;
-        const secondParan = lastUpdated.indexOf('-');
-        lastUpdated = parseInt(lastUpdated.slice(firstParan, secondParan));
-        return {
-          case: 'validData',
-          commStatus: busInfo.CommStatus,
-          delay: realtimeDelayData.delay,
-          destination: busInfo.Destination,
-          deviation: busInfo.Deviation,
-          direction: busInfo.Direction,
-          displayStatus: busInfo.DisplayStatus,
-          gpsStatus: busInfo.GPSStatus,
-          heading: busInfo.Heading,
-          lastStop: busInfo.LastStop,
-          lastUpdated,
-          latitude: busInfo.Latitude,
-          longitude: busInfo.Longitude,
-          name: busInfo.Name,
-          opStatus: busInfo.OpStatus,
-          routeID: busInfo.RouteId,
-          runID: busInfo.RunId,
-          speed: busInfo.Speed,
-          tripID: busInfo.TripId,
-          vehicleID: busInfo.VehicleId,
-        };
-      })(busFound);
+  await Promise.all(requestData.map((data) => {
```

Comment:
you don't need `Promise.all` here since you don't perform any async operations right?

Reply:
yes u r right